modules/cloud-guides/pages/integrate-teradata-jupyter-extensions-with-google-vertex-ai.adoc

```diff
@@ -0,0 +1,162 @@
+:page-revdate: May 12th, 2022
+:description: Integrate Teradata Jupyter extensions with Google Vertex AI
+:keywords: data warehouses, compute storage separation, teradata, vantage, cloud data platform, business intelligence, enterprise analytics, jupyter, teradatasql, ipython-sql, teradatasqlalchemy
+
+== Overview
+Teradata Jupyter extensions provide Teradata SQL kernel and several UI extensions to allow users to easily access and navigate Teradata database from Jupyter envioronment. Google Vertex AI is Google Cloud's new unified ML platform. Vertex AI Workbench provides a Jupyter-base development environment for the entire data science workflow. This article describes how to integate our Jupyter extensions with Vertex AI Workbench so that Vertex AI users can take advantage of our Teradata extensions in their ML pipeline.
+
+Vertex AI workbench supports two types of notebooks: managed notebooks and user-managed notebooks. Here we will focus on user-managed notebooks. We will show two ways to integrate our Jupyter extensions with user-managed notebooks: use startup script to install our kernel and extensions or use custom container.
+
+== Prerequisites
+* Access to a Teradata Vantage instance
+* Google Cloud account with Vertex AI enabled
+* Google cloud storage to store startup scripts and Teradata Jupyter extension package
+
+== Use startup script
+
+When we create a new notebook instance, we can specify a startup script. This script runs only once after the instance is created.
+
+=== 1. Download Teradata Jupyter extensions package
+Go to link:https://downloads.teradata.com/download/tools/vantage-modules-for-jupyter[Vantage Modules for Jupyter] page to download the Teradata Jupyter extensions package bundle Linux version.
+
+=== 2. Upload the package to a Google Cloud storage bucket.
+
+=== 3. Write a startup script and upload it to cloud storage bucket
+Below is a sample script. It fetches Teradata Jupyter extension package from cloud storage bucket and installs Teradata SQL kernel and extensions.
+
+[source, bash, role="content-editable"]
+----
+#! /bin/bash
+ 
+cd /home/jupyter
+mkdir teradata
+cd teradata
+gsutil cp gs://teradata-jupyter/* .
+unzip teradatasql*.zip
+ 
+# Install Teradata kernel
+cp teradatakernel /usr/local/bin
+ 
+jupyter kernelspec install ./teradatasql --prefix=/opt/conda
+ 
+# Install Teradata extensions
+jupyter labextension install --no-build teradata_database*
+jupyter labextension install --no-build teradata_connection_manager*
+jupyter labextension install --no-build teradata_resultset*
+jupyter labextension install --no-build teradata_sqlhighlighter*
+jupyter labextension install --no-build teradata_connection_manager*
+jupyter labextension install --no-build teradata_preferences*
+ 
+ 
+# PIP install the Teradata Python & R libraries
+pip install teradataml
+Rscript -e "install.packages('tdplyr',repos=c('https://teradata-download.s3.amazonaws.com','https://cloud.r-project.org'))"
```

Comment:
This line needs to change to point to the new Teradata R repo:

Rscript -e "install.packages('tdplyr',repos=c('https://r-repo.teradata.com','https://cloud.r-project.org'))"

Reply:
Also, as I'm thinking about it, if R and/or the R kernel is not installed in the Vertex AI environment this will not work.   Maybe not it as optional or leave it out.  Python and the Python kernel should always be there as part of Jupyter so leaving the teradataml is probably ok?

Reply:
fixed repo. commented this out by default and left a comment to uncomment it if user uses an environment that supports R.